pkg/sql/parser/sql.y

```diff
@@ -5677,81 +5681,60 @@ show_backup_stmt:
       InCollection:    $4.expr(),
     }
   }
-| SHOW BACKUP string_or_placeholder opt_with_options
-  {
-    $$.val = &tree.ShowBackup{
-      Details: tree.BackupDefaultDetails,
-      Path:    $3.expr(),
-      Options: $4.kvOptions(),
-    }
-  }
-| SHOW BACKUP string_or_placeholder IN string_or_placeholder opt_with_options
-  {
-    $$.val = &tree.ShowBackup{
-      Details: tree.BackupDefaultDetails,
-      Path:    $3.expr(),
-      InCollection: $5.expr(),
-      Options: $6.kvOptions(),
-    }
-  }
-| SHOW BACKUP SCHEMAS string_or_placeholder opt_with_options
+| SHOW BACKUP show_backup_kind string_or_placeholder IN string_or_placeholder opt_with_options
   {
     $$.val = &tree.ShowBackup{
-      Details: tree.BackupDefaultDetails,
-      ShouldIncludeSchemas: true,
+      Kind:    *$3.showBackupKind(),
       Path:    $4.expr(),
-      Options: $5.kvOptions(),
+      InCollection: $6.expr(),
+      Options: $7.kvOptions(),
     }
   }
-| SHOW BACKUP SCHEMAS string_or_placeholder IN string_or_placeholder opt_with_options
+| SHOW BACKUP show_backup_kind FROM string_or_placeholder IN string_or_placeholder opt_with_options
 	{
 		$$.val = &tree.ShowBackup{
-			Details: tree.BackupDefaultDetails,
-			ShouldIncludeSchemas: true,
-			Path:    $4.expr(),
-			InCollection: $6.expr(),
-			Options: $7.kvOptions(),
+			Kind:    *$3.showBackupKind(),
+			Path:    $5.expr(),
+			InCollection: $7.expr(),
+			Options: $8.kvOptions(),
 		}
 	}
-| SHOW BACKUP RANGES string_or_placeholder opt_with_options
-  {
-    /* SKIP DOC */
-    $$.val = &tree.ShowBackup{
-      Details: tree.BackupRangeDetails,
-      Path:    $4.expr(),
-      Options: $5.kvOptions(),
-    }
-  }
-| SHOW BACKUP RANGES string_or_placeholder IN string_or_placeholder opt_with_options
-  {
-		/* SKIP DOC */
+| SHOW BACKUP show_backup_kind string_or_placeholder opt_with_options
+	{
 		$$.val = &tree.ShowBackup{
-			Details: tree.BackupRangeDetails,
+			Kind:    *$3.showBackupKind(),
 			Path:    $4.expr(),
-			InCollection: $6.expr(),
-			Options: $7.kvOptions(),
+			Options: $5.kvOptions(),
 		}
-  }
-| SHOW BACKUP FILES string_or_placeholder opt_with_options
+	}
+| SHOW BACKUP error // SHOW HELP: SHOW BACKUP
+
+show_backup_kind:
+  /* EMPTY -- default */
```

Comment:
This seems like it'd be ambiguous since `files` is a non-reserved word too, so you could match it here, or not match it here and match it with string_or_placeholder? 

I wonder if we need to require `FROM` for and only for the modified kind case?

so 
```
 SHOW BACKUP string_or_placeholder opt_in_collection opt_with_options
...
| SHOW BACKUP backup_property FROM string_or_placehoder opt_in_collection opt_with_options
``` ?

Reply:
aha! nice catch. Just implemented this and it solved my build problems!